2214/2/battle_expert.rb

```diff
@@ -0,0 +1,94 @@
+require 'commander'
+require 'terminal-table'
+require_relative 'bad_words_counter'
+require_relative 'words_in_round_counter'
+require_relative 'popular_words_counter'
+require_relative 'battler'
+require_relative 'battle'
+
+class BattleExpert
+  BATTLES_FOLDER = 'Battles'.freeze
+  def initialize
+    @battles = take_battles
+    @battlers_names = find_all_battlers
+  end
+
+  def describe_battlers(top_bad_words)
+    battlers = []
+    @battlers_names.each { |battler_name| battlers << process_battler(battler_name) }
+    battlers = sort_battlers(battlers)
+    show_top_battlers(battlers, top_bad_words)
+  end
+
+  def find_popular_words(battler_name, top_words = 30)
+    if @battlers_names.include? battler_name
+      PopularWordsCounter.count(@battles, battler_name, top_words)
+    else
+      puts "Я не знаю МЦ #{battler_name}. Зато мне известны:"
+      @battlers_names.each { |battler| puts battler }
+    end
+  end
+
+  private
+
+  def process_battler(battler_name)
+    count_of_battles = count_battles(battler_name)
+    bad_words = BadWordsCounter.count(@battles, battler_name)
+    words_in_round = WordsInRoundCounter.count(@battles, battler_name, count_of_battles)
+    average_number = (bad_words.to_f / count_of_battles).round(2)
+    Battler.new(battler_name, count_of_battles, bad_words, average_number, words_in_round)
```

Comment:
This method and others do a lot of work in them. I think we should organize the code better. For example, you can move code that counts statistics for batters at least to the `Battler` class:

```rb
class Battler
  def initialize(name, battles)
    # ...
  end

  def count_of_battles
    # ...
  end

  def bad_words_per_round
    # ...
  end

  def average_number_of_words
    # ...
  end
end
```

So the `BattleExpert` don't worry about those calculations.